src/engine/worker.cpp

```diff
@@ -18,8 +19,59 @@
 #include "sampling/logits_processor.h"
 #include "sampling/sampler.h"
 
+DEFINE_bool(enable_cudagraph, false, "Enable CUDAGraph to optimize model execution.");
+
 namespace llm {
 
+class CudaGraphRunner {
+ public:
+  explicit CudaGraphRunner(CausalLM* model) : model_(model) {}
+  ~CudaGraphRunner() = default;
+
+  // ids, positions, kv_cache, input_meta, memorypool
+  void capture(torch::Tensor flatten_tokens, torch::Tensor flatten_positions,
+      const InputParameters& params, std::vector<KVCache>& kv_cache) {
+    // run model once to avoid captured graph does not include initial benchmarking
+    model_->forward(flatten_tokens,
+                    flatten_positions,
+                    kv_cache,
+                    params);
+    torch::cuda::synchronize();
+
+    // create cudagraph and capture
+    graph_ = std::make_unique<at::cuda::CUDAGraph>();
+    graph_->capture_begin();
+    auto logits = model_->forward(flatten_tokens,
```

Comment:
the logits is var len, depends on the number of seqs in the batch. not sure if graph support this.